source/default_mode/WorldSetup.cc

```diff
@@ -2,71 +2,68 @@
+ */
+void SymWorld::SetupHosts(long unsigned int* POP_SIZE){ 
+  for (size_t i = 0; i < *POP_SIZE; i++) {
+  emp::Ptr<Host> new_org;
+  //since competition is checked first, don't  need the && !competition part of  rand host phenotype
+  new_org.New(&GetRandom(), this, my_config, my_config->HOST_INT());
+  InjectHost(new_org);
+  }
+
+}
 
-void worldSetup(emp::Ptr<SymWorld> world, emp::Ptr<SymConfigBase> my_config) {
-// params
-  emp::Random& random = world->GetRandom();
 
+/**
+ * Input: The number of symbionts and whether the symbionts have random phenotypes.
+ *
+ * Output: None.
+ *
+ * Purpose: To populate the world with symbionts with appropriate phenotypes.
+ */
+void SymWorld::SetupSymbionts(long unsigned int *total_syms) {
+  //This loop must be outside of the host generation loop since otherwise
+  //syms try to inject into mostly empty spots at first
+  for (size_t j = 0; j < *total_syms; j++) {
+    emp::Ptr<Symbiont> new_sym = emp::NewPtr<Symbiont>(&GetRandom(), this, my_config, my_config->SYM_INT(), 0);
+    InjectSymbiont(new_sym);
+  }
+}
+
+/**
+ * Input: None.
+ *
+ * Output: None.
+ *
+ * Purpose: Prepare the world for a simulation by applying the configuration settings 
+ * and populating the world with hosts and symbionts.
+ */
+void SymWorld::Setup() {
   double start_moi = my_config->START_MOI();
   long unsigned int POP_SIZE;
   if (my_config->POP_SIZE() == -1) {
     POP_SIZE = my_config->GRID_X() * my_config->GRID_Y();
   } else {
     POP_SIZE = my_config->POP_SIZE();
   }
-  bool random_phen_host = false;
-  bool random_phen_sym = false;
-  if(my_config->HOST_INT() == -2 && !my_config->COMPETITION_MODE()) random_phen_host = true;
-  if(my_config->SYM_INT() == -2) random_phen_sym = true;
-
-  if (my_config->GRID() == 0) {world->SetPopStruct_Mixed(false);}
-  else world->SetPopStruct_Grid(my_config->GRID_X(), my_config->GRID_Y(), false);
-
-
-  double comp_host_1 = 0;
-  double comp_host_2 = 0.95;
-
-
-
-  //inject hosts
-  for (size_t i = 0; i < POP_SIZE; i++){
-    emp::Ptr<Host> new_org;
-
-    if (random_phen_host) {new_org.New(&random, world, my_config, random.GetDouble(-1, 1));
-    } else if (my_config->COMPETITION_MODE() && i%2==0) {
-        new_org.New(&random, world, my_config, comp_host_1);
-    } else if (my_config->COMPETITION_MODE() && i%2==1) {
-        new_org.New(&random, world, my_config, comp_host_2);
-    } else { new_org.New(&random, world, my_config, my_config->HOST_INT());
-    }
-    //Currently hacked because there isn't an AddOrg function, but there probably should be
-    if(my_config->GRID()) {
-      world->AddOrgAt(new_org, emp::WorldPosition(world->GetRandomCellID()));
-    } else {
-      world->AddOrgAt(new_org, world->size());
-    }
-    //world.Inject(*new_org);
-  }
 
-  //sets up the world size
-  world->Resize(my_config->GRID_X(), my_config->GRID_Y());
+  if (my_config->GRID() == 0) {SetPopStruct_Mixed(false);}
```

Comment:
Let's put in a comment explaining that the 'false' isn't saying that we aren't setting it to Mixed, this always confuses me. The 'false' is about there not being synchronous generations. It's a keyword argument, so we might be able to actually just put synchronous_gen=false as the parameter to make it more clear?